pkg/metrics/delivery_reporter.go

```diff
@@ -120,15 +123,25 @@ func NewDeliveryReporter(podName PodName, containerName ContainerName) (*Deliver
 	return r, nil
 }
 
+func WithRespStatusCode(ctx context.Context, responseCode int) context.Context {
+	return context.WithValue(ctx, respStatusCode, responseCode)
```

Comment:
This can be simplified by just inserting the relevant tags into context.

Reply:
done